### PR TITLE
Solved #325 Adding retry mechanism

### LINKED ISSUE #325: Adding retry mechanism code for the database connection
# Description

Sometimes errors can occur when you are connecting to a Database, because network connections to a database are never 100% perfect. Therefore i would like to add a python script in which the retry mechanism will be implemented so that if suppose first time it fails to connect with the DB then at least 3 times it should try reconnecting with the database.

## Language

- [x] Python

## Type of issue

_Please delete options that are not relevant._

- [x] Feature

## Checklist:

- **For Feature**

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| Feature request |Adding retry mechanism code for the database connection. | feature, CH-20, medium | self-assign |

### PR DESCRIPTION
**Pull request by sameersrivastava13**
# Description
Added a python script in which the retry mechanism will be implemented so that if suppose first time it fails to connect with the DB then at least 3 times it should try reconnecting with the database.

Language

Fixes #325 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sameersrivastava13** [comment]
@srimani-programmer @HarshCasper Please review.

File: Python/Aws/retryMechanism.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from pg import DB, InternalError
```
Comment (srimani-programmer) [inline Python/Aws/retryMechanism.py:1]:
```suggestion
What is this pg module?
```

**Review by srimani-programmer — CHANGES_REQUESTED** [review]
Please Add the proper comments and README file for the project. Please explain what is the pg module? let me know. If any dependencies are there for the project please add it in the project README file.

File: Python/Aws/retryMechanism.py (hunk 1 above)
Comment (sameersrivastava13) [reply Python/Aws/retryMechanism.py:1]: `pg`  refers to the `PyGreSQL` driver. `PyGreSQL` is an open-source Python module that interfaces to a PostgreSQL database. It embeds the PostgreSQL query library to allow easy use of the powerful PostgreSQL features from a Python script.

For more information read [this](http://www.pygresql.org/)

**Comment by sameersrivastava13** [comment]
> Please Add the proper comments and README file for the project. Please explain what is the pg module? let me know. If any dependencies are there for the project please add it in the project README file.

Okay sure I'll add the readme file and proper comments in the code. But here the main purpose of the issue was how we can actually retry connecting to the database. Whether its a mysql or postgresql database or any other.  The logic will be the same.

**Comment by sameersrivastava13** [comment]
@HarshCasper Can you please review.

**Review by srimani-programmer — CHANGES_REQUESTED** [review]
What about the other databases if it only support postgres? please work on the remaining databases as well.

**Comment by sameersrivastava13** [comment]
> What about the other databases if it only support postgres? please work on the remaining databases as well.

@srimani-programmer try and understand. It will work for all the databases. Just to give an example i have used `PostgreSQL`. Please look at the logic part (the while loop) that is the part of the code where the retry mechanism is actually implemented. Doesn't matter if you are using `PostgreSQL` or `MySQL`. When you connect with the database, if the `Connection object` associated with `db` is empty we will retry connecting if it is not empty then simply we can query the database directly using raw SQL statements. If it is still not clear we can discuss this on slack.

**Pull request merged**